### PR TITLE
Enable Fused Kernels by Default for Memory Efficiency

### PR DESCRIPTION
**Pull request by 0hujun**
## Summary

This PR proposes changing the default value of `use_fused_kernels` from `False` to `True` across all engine backends (FSDP2, Megatron, VeOmni, AutoModel). Fused kernels provide significant memory savings (32x reduction in logits memory) and enable longer context training without sacrificing correctness. The change includes graceful fallback logic for incompatible configurations.

### The Problem

Currently, `use_fused_kernels` defaults to `False` in three config classes:

| Config Class    | File                            | Line | Default |
| --------------- | ------------------------------- | ---- | ------- |
| `HFModelConfig` | `verl/workers/config/model.py`  | 138  | `False` |
| `ActorConfig`   | `verl/workers/config/actor.py`  | 182  | `False` |
| `EngineConfig`  | `verl/workers/config/engine.py` | 110  | `False` |

And in the canonical YAML config:

```yaml
# verl/trainer/config/model/hf_model.yaml
use_fused_kernels: False
```

This means users must explicitly opt-in to fused kernels, and most users are unaware of this feature, resulting in:

1. **Unnecessary OOM errors** at long context lengths (32K+)
2. **Suboptimal memory usage** — full logits materialization wastes ~30 GB of GPU memory
3. **Shorter maximum context** — users hit memory limits earlier than necessary

### The Benefits

Fused kernels avoid materializing the full logits tensor by computing log_probs and entropy directly from hidden_states and vocab_weights:

| Component              | Without Fused Kernels | With Fused Kernels | Reduction  |
| ---------------------- | --------------------- | ------------------ | ---------- |
| Logits (bf16)          | 9.93 GB               | 0.31 GB            | **32x**    |
| Logits gradient (fp32) | 19.87 GB              | 0.31 GB            | **64x**    |
| **Total peak savings** | —                     | —                  | **~29 GB** |

Real-world impact (Qwen3.5-9B, 32K context, Ascend910 61GB NPU):

- Without fused kernels: OOM at 32K context
- With fused kernels: trains successfully at 32K, peak memory 34.4 GB (56%)

### CODE REVIEW

**Review by Luosuu — DISMISSED** [review]

**Comment by Luosuu** [comment]
@0hujun please fix pre-commit
